File: src/Models/db/lyda/Album.ts
import {User} from "./User";
import {AlbumTrack} from "./AlbumTrack";
import {Entity} from "@targoninc/ts-search";
import {InteractionMetadata} from "../../InteractionMetadata";
import {TrackLike} from "./TrackLike";
import {Visibility} from "../../../Enums/Visibility";

export interface Album extends Entity {
    tracks?: AlbumTrack[];
    user?: User;
    id: number;
    user_id: number;
    title: string;
    description: string;
    upc: string;
    release_date: Date;
    created_at: Date;
    updated_at: Date;
    visibility: Visibility;
    secretcode: string;
    has_cover: boolean;
    price: number;
    likes?: InteractionMetadata<TrackLike>;
    /*reposts?: InteractionMetadata<Repost>;
    comments?: InteractionMetadata<Comment>;*/
}

File: kotlin/com/example/lydaandroid/model/Album.kt
package com.example.lydaandroid.model

import java.util.Date

data class Album(
    val id: Int? = null,
    val user_id: Int? = null,
    val title: String = "",
    val description: String = "",
    val upc: String = "",
    val release_date: String = "",
    val created_at: String = "",
    val updated_at: String = "",
    val visibility: String? = "public",
    val secretcode: String? = null,
    val has_cover: Boolean = false,
    val price: Double = 0.0,
    val likes: TrackLikeMetadata? = null,
    val user: User? = null
)